Competition2018/src/main/java/competition/TiltPreventionModule.java

```diff
@@ -0,0 +1,35 @@
+package competition;
+
+import xbot.common.properties.DoubleProperty;
+import xbot.common.properties.XPropertyManager;
+import com.google.inject.Inject;
+import competition.subsystems.pose.PoseSubsystem;
+
+
+public class TiltPreventionModule{
+
+    PoseSubsystem pose;
+    double goodPoint;
+    double goodHeading;
+    DoubleProperty pitchThrehold;
+
+
+    @Inject
+    public TiltPreventionModule(PoseSubsystem pose, XPropertyManager propManager) {
+        this.pose = pose;
+        pitchThrehold = propManager.createPersistentProperty("Absolute Value of Pitch Threhold", 10);
+    }
+
+    public double preventTilt (double potentialPower) {
+
+        if (pose.getRobotPitch() > pitchThrehold.get() && potentialPower < 0) {
```

Comment:
Let's test on the robot if these are the right polarity - I _think_ positive pitch means the robot is tipping backwards, meaning we'd want to only allow driving backwards, but I'm not sure. Let's check to be sure.

Reply:
Okay! I will put "needs robot" on Asana!